tools/db_stress.cc

```diff
@@ -1823,10 +1845,27 @@ class StressTest {
       write_opts.sync = true;
     }
     write_opts.disableWAL = FLAGS_disable_wal;
-    const int prefixBound = (int)FLAGS_readpercent + (int)FLAGS_prefixpercent;
-    const int writeBound = prefixBound + (int)FLAGS_writepercent;
-    const int delBound = writeBound + (int)FLAGS_delpercent;
-    const int delRangeBound = delBound + (int)FLAGS_delrangepercent;
+
+    int read_bound, prefix_bound, write_bound, del_bound, del_range_bound,
+        iter_bound;
+    if (FLAGS_readweight >= 0) {
+      // If one weight is specified all were specified (enforced in options
+      // validation code).
+      read_bound = FLAGS_readweight;
+      prefix_bound = read_bound + FLAGS_prefixweight;
+      write_bound = prefix_bound + FLAGS_writeweight;
+      del_bound = write_bound + FLAGS_delweight;
+      del_range_bound = del_bound + FLAGS_delrangeweight;
+      iter_bound = del_range_bound + FLAGS_iterweight;
```

Comment:
Right. I wanted to be able to specify arbitrary ratios because `DeleteRange` is really slow in some scenarios (actually I already found 1/10000 is too high for the CI test to work well). I'm not sure this is a good interface, but it at least avoided floating-point arithmetic, which was nice. Let me know if you have any alternative suggestion.

Reply:
SGTM